samples/client.c

```diff
@@ -535,15 +536,23 @@ set_migration_state(int sock, uint32_t state)
             .iov_len = sizeof(change_state)
         }
     };
+    void* response = malloc(sizeof(req) + sizeof(change_state));
     
+    pthread_mutex_lock(&mutex);
     int ret = tran_sock_msg_iovec(sock, msg_id--, VFIO_USER_DEVICE_FEATURE,
                                   send_iovecs, 3, NULL, 0, NULL,
-                                  &req, sizeof(req), NULL, 0);
+                                  response, sizeof(req) + sizeof(change_state),
```

Comment:
This change should go into the earlier commit that touched this code for migration v2.